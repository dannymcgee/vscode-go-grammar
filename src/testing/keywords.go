import "foo"

if x > max {
	x = max
}

if x := f(); x < y {
	return x
} else if x > z {
	return z
} else {
	return y
}

switch tag {
default: s3()
case 0, 1, 2, 3: s1()
case 4, 5, 6, 7: s2()
}

switch x := f(); {
case x < 0: return -x
default: return x
}

switch {
case x < y: f1()
case x < z: f2()
case x == 4: f3()
}

for a < b {
	a *= 2
}

for i := 0; i < 10; i++ {}

for i, _ := range testdata.a {}

var a [10]string
for i, s := range a {}

var key string
var val interface{}

m := map[string]int{"foo":0}
for key, val = range m {
	h(key, val)
}

var ch chan Work = producer()
for w := range ch {
	doWork(w)
}

// empty a channel
for range ch {}

go Server()
go func(ch chan<- bool) { for { sleep(10); ch <- true }} (c)
