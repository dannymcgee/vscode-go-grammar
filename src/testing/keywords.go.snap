// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`keywords should work 1`] = `

-> import "foo"
 | ^^^^^^ keyword.control.import.go
 |        ^ punctuation.definition.string.begin.go  string.interpreted.go
 |         ^^^ string.interpreted.go
 |            ^ punctuation.definition.string.end.go  string.interpreted.go
-> 
-> if x > max {
 | ^^ keyword.control.if.go
 |    ^ variable.other.go
 |      ^ keyword.operator.comparison.>.go
 |        ^^^ variable.other.go
 |            ^ punctuation.brace.curly.go
->     x = max
 |     ^ variable.other.go
 |       ^ keyword.operator.assignment.go
 |         ^^^ variable.other.go
-> }
 | ^ punctuation.brace.curly.go
-> 
-> if x := f(); x < y {
 | ^^ keyword.control.if.go
 |    ^ variable.other.go
 |      ^^ keyword.operator.assignment.go
 |         ^ entity.name.function.go
 |          ^ punctuation.brace.round.go
 |           ^ punctuation.brace.round.go
 |            ^ punctuation.terminator.go
 |              ^ variable.other.go
 |                ^ keyword.operator.comparison.<.go
 |                  ^ variable.other.go
 |                    ^ punctuation.brace.curly.go
->     return x
 |     ^^^^^^ keyword.control.return.go
 |            ^ variable.other.go
-> } else if x > z {
 | ^ punctuation.brace.curly.go
 |   ^^^^ keyword.control.else.go
 |        ^^ keyword.control.if.go
 |           ^ variable.other.go
 |             ^ keyword.operator.comparison.>.go
 |               ^ variable.other.go
 |                 ^ punctuation.brace.curly.go
->     return z
 |     ^^^^^^ keyword.control.return.go
 |            ^ variable.other.go
-> } else {
 | ^ punctuation.brace.curly.go
 |   ^^^^ keyword.control.else.go
 |        ^ punctuation.brace.curly.go
->     return y
 |     ^^^^^^ keyword.control.return.go
 |            ^ variable.other.go
-> }
 | ^ punctuation.brace.curly.go
-> 
-> switch tag {
 | ^^^^^^ keyword.control.switch.go
 |        ^^^ variable.other.go
 |            ^ punctuation.brace.curly.go
-> default: s3()
 | ^^^^^^^ keyword.control.default.go
 |        ^ punctuation.separator.key-value.go
 |          ^^ entity.name.function.go
 |            ^ punctuation.brace.round.go
 |             ^ punctuation.brace.round.go
-> case 0, 1, 2, 3: s1()
 | ^^^^ keyword.control.case.go
 |      ^ constant.numeric.integer.go
 |       ^ punctuation.separator.comma.go
 |         ^ constant.numeric.integer.go
 |          ^ punctuation.separator.comma.go
 |            ^ constant.numeric.integer.go
 |             ^ punctuation.separator.comma.go
 |               ^ constant.numeric.integer.go
 |                ^ punctuation.separator.key-value.go
 |                  ^^ entity.name.function.go
 |                    ^ punctuation.brace.round.go
 |                     ^ punctuation.brace.round.go
-> case 4, 5, 6, 7: s2()
 | ^^^^ keyword.control.case.go
 |      ^ constant.numeric.integer.go
 |       ^ punctuation.separator.comma.go
 |         ^ constant.numeric.integer.go
 |          ^ punctuation.separator.comma.go
 |            ^ constant.numeric.integer.go
 |             ^ punctuation.separator.comma.go
 |               ^ constant.numeric.integer.go
 |                ^ punctuation.separator.key-value.go
 |                  ^^ entity.name.function.go
 |                    ^ punctuation.brace.round.go
 |                     ^ punctuation.brace.round.go
-> }
 | ^ punctuation.brace.curly.go
-> 
-> switch x := f(); {
 | ^^^^^^ keyword.control.switch.go
 |        ^ variable.other.go
 |          ^^ keyword.operator.assignment.go
 |             ^ entity.name.function.go
 |              ^ punctuation.brace.round.go
 |               ^ punctuation.brace.round.go
 |                ^ punctuation.terminator.go
 |                  ^ punctuation.brace.curly.go
-> case x < 0: return -x
 | ^^^^ keyword.control.case.go
 |      ^ variable.other.go
 |        ^ keyword.operator.comparison.<.go
 |          ^ constant.numeric.integer.go
 |           ^ punctuation.separator.key-value.go
 |             ^^^^^^ keyword.control.return.go
 |                    ^ keyword.operator.arithmetic.-.go
 |                     ^ variable.other.go
-> default: return x
 | ^^^^^^^ keyword.control.default.go
 |        ^ punctuation.separator.key-value.go
 |          ^^^^^^ keyword.control.return.go
 |                 ^ variable.other.go
-> }
 | ^ punctuation.brace.curly.go
-> 
-> switch {
 | ^^^^^^ keyword.control.switch.go
 |        ^ punctuation.brace.curly.go
-> case x < y: f1()
 | ^^^^ keyword.control.case.go
 |      ^ variable.other.go
 |        ^ keyword.operator.comparison.<.go
 |          ^ variable.field.go
 |           ^ punctuation.separator.key-value.go
 |             ^^ entity.name.function.go
 |               ^ punctuation.brace.round.go
 |                ^ punctuation.brace.round.go
-> case x < z: f2()
 | ^^^^ keyword.control.case.go
 |      ^ variable.other.go
 |        ^ keyword.operator.comparison.<.go
 |          ^ variable.field.go
 |           ^ punctuation.separator.key-value.go
 |             ^^ entity.name.function.go
 |               ^ punctuation.brace.round.go
 |                ^ punctuation.brace.round.go
-> case x == 4: f3()
 | ^^^^ keyword.control.case.go
 |      ^ variable.other.go
 |        ^^ keyword.operator.comparison.==.go
 |           ^ constant.numeric.integer.go
 |            ^ punctuation.separator.key-value.go
 |              ^^ entity.name.function.go
 |                ^ punctuation.brace.round.go
 |                 ^ punctuation.brace.round.go
-> }
 | ^ punctuation.brace.curly.go
-> 
-> for a < b {
 | ^^^ keyword.control.for.go
 |     ^ variable.other.go
 |       ^ keyword.operator.comparison.<.go
 |         ^ variable.other.go
 |           ^ punctuation.brace.curly.go
->     a *= 2
 |     ^ variable.other.go
 |       ^^ keyword.operator.arithmetic.*=.go
 |          ^ constant.numeric.integer.go
-> }
 | ^ punctuation.brace.curly.go
-> 
-> for i := 0; i < 10; i++ {}
 | ^^^ keyword.control.for.go
 |     ^ variable.other.go
 |       ^^ keyword.operator.assignment.go
 |          ^ constant.numeric.integer.go
 |           ^ punctuation.terminator.go
 |             ^ variable.other.go
 |               ^ keyword.operator.comparison.<.go
 |                 ^^ constant.numeric.integer.go
 |                   ^ punctuation.terminator.go
 |                     ^ variable.other.go
 |                      ^^ keyword.operator.arithmetic.++.go
 |                         ^ punctuation.brace.curly.go
 |                          ^ punctuation.brace.curly.go
-> 
-> for i, _ := range testdata.a {}
 | ^^^ keyword.control.for.go
 |     ^ variable.other.go
 |      ^ punctuation.separator.comma.go
 |        ^ variable.other.go
 |          ^^ keyword.operator.assignment.go
 |             ^^^^^ keyword.operator.expression.range.go
 |                   ^^^^^^^^ variable.other.go
 |                           ^ punctuation.accessor.go
 |                            ^ variable.field.go
 |                              ^ punctuation.brace.curly.go
 |                               ^ punctuation.brace.curly.go
-> 
-> var a [10]string
 | ^^^ storage.type.var.go
 |     ^ variable.other.go
 |       ^ punctuation.brace.square.go
 |        ^^ constant.numeric.integer.go
 |          ^ punctuation.brace.square.go
 |           ^^^^^^ support.type.primitive.go
-> for i, s := range a {}
 | ^^^ keyword.control.for.go
 |     ^ variable.other.go
 |      ^ punctuation.separator.comma.go
 |        ^ variable.other.go
 |          ^^ keyword.operator.assignment.go
 |             ^^^^^ keyword.operator.expression.range.go
 |                   ^ variable.other.go
 |                     ^ punctuation.brace.curly.go
 |                      ^ punctuation.brace.curly.go
-> 
-> var key string
 | ^^^ storage.type.var.go
 |     ^^^ variable.other.go
 |         ^^^^^^ support.type.primitive.go
-> var val interface{}
 | ^^^ storage.type.var.go
 |     ^^^ variable.other.go
 |         ^^^^^^^^^ storage.type.interface.go
 |                  ^ punctuation.brace.curly.go
 |                   ^ punctuation.brace.curly.go
-> 
-> m := map[string]int{"foo":0}
 | ^ variable.other.go
 |   ^^ keyword.operator.assignment.go
 |      ^^^ storage.type.map.go
 |         ^ punctuation.brace.square.go
 |          ^^^^^^ support.type.primitive.go
 |                ^ punctuation.brace.square.go
 |                 ^^^ support.type.primitive.go
 |                    ^ punctuation.brace.curly.go
 |                     ^ punctuation.definition.string.begin.go  string.interpreted.go
 |                      ^^^ string.interpreted.go
 |                         ^ punctuation.definition.string.end.go  string.interpreted.go
 |                          ^ punctuation.separator.key-value.go
 |                           ^ constant.numeric.integer.go
 |                            ^ punctuation.brace.curly.go
-> for key, val = range m {
 | ^^^ keyword.control.for.go
 |     ^^^ variable.other.go
 |        ^ punctuation.separator.comma.go
 |          ^^^ variable.other.go
 |              ^ keyword.operator.assignment.go
 |                ^^^^^ keyword.operator.expression.range.go
 |                      ^ variable.other.go
 |                        ^ punctuation.brace.curly.go
->     h(key, val)
 |     ^ entity.name.function.go
 |      ^ punctuation.brace.round.go
 |       ^^^ variable.other.go
 |          ^ punctuation.separator.comma.go
 |            ^^^ variable.other.go
 |               ^ punctuation.brace.round.go
-> }
 | ^ punctuation.brace.curly.go
-> 
-> var ch chan Work = producer()
 | ^^^ storage.type.var.go
 |     ^^ variable.other.go
 |        ^^^^ storage.type.chan.go
 |             ^^^^ entity.name.type.go
 |                  ^ keyword.operator.assignment.go
 |                    ^^^^^^^^ entity.name.function.go
 |                            ^ punctuation.brace.round.go
 |                             ^ punctuation.brace.round.go
-> for w := range ch {
 | ^^^ keyword.control.for.go
 |     ^ variable.other.go
 |       ^^ keyword.operator.assignment.go
 |          ^^^^^ keyword.operator.expression.range.go
 |                ^^ variable.other.go
 |                   ^ punctuation.brace.curly.go
->     doWork(w)
 |     ^^^^^^ entity.name.function.go
 |           ^ punctuation.brace.round.go
 |            ^ variable.other.go
 |             ^ punctuation.brace.round.go
-> }
 | ^ punctuation.brace.curly.go
-> 
-> // empty a channel
 | ^^ punctuation.definition.comment.line.go  comment.line.go
 |   ^^^^^^^^^^^^^^^^ comment.line.go
-> for range ch {}
 | ^^^ keyword.control.for.go
 |     ^^^^^ keyword.operator.expression.range.go
 |           ^^ variable.other.go
 |              ^ punctuation.brace.curly.go
 |               ^ punctuation.brace.curly.go
-> 
-> go Server()
 | ^^ keyword.control.go.go
 |    ^^^^^^ entity.name.function.go
 |          ^ punctuation.brace.round.go
 |           ^ punctuation.brace.round.go
-> go func(ch chan <- bool) { for { sleep(10); ch <- true }} (c)
 | ^^ keyword.control.go.go
 |    ^^^^ storage.type.func.go
 |        ^ punctuation.definition.parameters.begin.go
 |         ^^ variable.other.go
 |            ^^^^ storage.type.chan.go
 |                 ^^ keyword.operator.arrow.go
 |                    ^^^^ support.type.primitive.go
 |                        ^ punctuation.definition.parameters.end.go
 |                          ^ punctuation.definition.block.begin.go
 |                            ^^^ keyword.control.for.go
 |                                ^ punctuation.brace.curly.go
 |                                  ^^^^^ entity.name.function.go
 |                                       ^ punctuation.brace.round.go
 |                                        ^^ constant.numeric.integer.go
 |                                          ^ punctuation.brace.round.go
 |                                           ^ punctuation.terminator.go
 |                                             ^^ variable.other.go
 |                                                ^^ keyword.operator.arrow.go
 |                                                   ^^^^ constant.language.boolean.go
 |                                                        ^ punctuation.brace.curly.go
 |                                                         ^ punctuation.brace.curly.go
 |                                                           ^ punctuation.brace.round.go
 |                                                            ^ variable.other.go
 |                                                             ^ punctuation.brace.round.go
-> 
`;
